tutorials/filter/src/test/java/com/datatorrent/tutorial/filter/ApplicationTest.java

```diff
@@ -0,0 +1,109 @@
+package com.datatorrent.tutorial.filter;
+/**
+ * Copyright (c) 2016 DataTorrent, Inc.
+ * All rights reserved.
+ */
+
+import java.io.File;
+import java.io.IOException;
+
+import javax.validation.ConstraintViolationException;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.conf.Configuration;
+
+import com.datatorrent.api.LocalMode;
+
+/**
+ * Test the DAG declaration in local mode.
+ */
+public class ApplicationTest
+{
+  private String outputDir;
+  
+  public static class TestMeta extends TestWatcher
+  {
+    public String baseDirectory;
+
+    @Override
+    protected void starting(org.junit.runner.Description description)
+    {
+      this.baseDirectory = "target/" + description.getClassName() + "/" + description.getMethodName();
+    }
+    
+    @Override
+    protected void finished(Description description)
+    {
+      super.finished(description);
+      try {
+        FileUtils.forceDelete(new File(baseDirectory));
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+
+  }
+  
+  @Rule
+  public TestMeta testMeta = new TestMeta();
+  
+  @Before
+  public void setup() throws Exception
+  {
+    outputDir = testMeta.baseDirectory + File.separator + "output";
+  }
+  
+  @Test
+  public void testApplication() throws IOException, Exception
+  {
+
+    try {
+      LocalMode lma = LocalMode.newInstance();
+      Configuration conf = new Configuration(false);
+      conf.addResource(this.getClass().getResourceAsStream("/META-INF/properties.xml"));
+      conf.set("dt.application.FilterExample.operator.selectedOutput.prop.filePath", outputDir);
+      conf.set("dt.application.FilterExample.operator.rejectedOutput.prop.filePath", outputDir);
+      File selectedfile = FileUtils.getFile(outputDir, "selected.txt_6.0");
```

Comment:
It is based on the operatorId which is derived from the order in which operators are added to the DAG. Thus, it is predictable. I tried with another approach for looking for files which begins with "selected.txt". But it will also include .tmp files which are not final. Finalized files are not available before we start the lc.runAsync(). Thus, IMO the current implementation should be fine.